src/App.tsx

```diff
@@ -1,93 +1,202 @@
+  const [newError, setNewError] = useState<ErrorMessages | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [deleteTodosId, setDeleteTodosId] = useState<number[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const isTodoShow = todos.length > 0;
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setNewError(ErrorMessages.LoadError);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const deleteTodo = async (todoId: number) => {
+    try {
+      setDeleteTodosId((prev) => [...prev, todoId]);
+
+      const removeNewTodo = await removeTodo(todoId);
+
+      if (!removeNewTodo) {
+        setNewError(ErrorMessages.DeleteError);
+
+        return;
+      }
+
+      setTodos((prev) => {
+        const filterPrev = prev.filter((todo) => todo.id !== todoId);
+
+        return [...filterPrev];
+      });
+    } catch (error) {
+      setNewError(ErrorMessages.DeleteError);
+    } finally {
+      setDeleteTodosId([]);
+    }
+  };
+
+  const filterTodos = useMemo(() => {
+    return getFilteredTodos(todos, filterBy);
+  }, [todos, filterBy]);
+
+  const clearTodoTitle = () => {
+    setNewTodoTitle('');
+  };
+
+  const addNewTodo = async (title: string): Promise<null | Todo> => {
+    if (!title) {
+      setNewError(ErrorMessages.TitleError);
+
+      return null;
+    }
+
+    try {
+      const newTodoStr = {
+        completed: false,
+        title,
+        userId: USER_ID,
+      };
+
+      const newTodo = await createTodo(newTodoStr);
+
+      clearTodoTitle();
+
+      setTodos((prev) => [...prev, newTodo]);
+
+      return newTodo;
+    } catch (error) {
+      setNewError(ErrorMessages.TitleError);
+
+      return null;
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const updateTodoItem = async (todoId: number, updatedTodo: Todo) => {
+    try {
+      setNewError(null);
+      await client.patch<Todo>(`/todos/${todoId}`, updatedTodo);
+
+      setTodos(prevTodos => prevTodos.map(
+        todo => (todo.id === todoId ? updatedTodo : todo),
+      ));
+    } catch (error) {
+      setNewError(ErrorMessages.UpdateError);
+    }
+  };
+
+  const handleAll = () => {
+    const allCompleted = todos.every(todo => todo.completed);
+
+    const updateTodos = todos.map(todo => ({
+      ...todo,
+      completed: !allCompleted,
+    }));
+
+    setTodos(updateTodos);
+  };
+
+  const handleChackBox = async (updateNewTodo: Todo): Promise<void> => {
+    try {
+      const newTodo = {
+        ...updateNewTodo,
+        completed: !updateNewTodo.completed,
+      };
+
+      await updateTodo(newTodo);
+
+      setTodos((current) => current
+        .map((todo) => (todo.id === newTodo.id ? newTodo : todo)));
+    } catch (error) {
+      setNewError(ErrorMessages.TitleError);
```

Comment:
wrong error type
Also, the error appears w/o any reason:
[Screencast from 21-09-23 21:01:47.webm](https://github.com/mate-academy/react_todo-app/assets/86923089/c16407ab-a7cd-4ee2-9f6f-2dffd6192072)